zokrates_core/src/typed_absy/result_folder.rs

```diff
@@ -800,15 +807,22 @@ pub fn fold_struct_expression<'ast, T: Field, F: ResultFolder<'ast, T>>(
     })
 }
 
+pub fn fold_constant<'ast, T: Field, F: ResultFolder<'ast, T>>(
+    f: &mut F,
+    c: TypedConstant<'ast, T>,
+) -> Result<TypedConstant<'ast, T>, F::Error> {
+    Ok(TypedConstant {
+        expression: f.fold_expression(c.expression)?,
+        ..c
```

Comment:
I think the type also needs to be folded: array sizes can be any expression, including a constant

Reply:
Or the type needs to be removed from `TypedConstant`: isn't it always `expression.get_type()` ?

Reply:
Nice catch, it is! Does it even make sense then to have `TypedConstant` struct instead of just using `TypedExpression` directly?

Reply:
I think it doesn't hurt

Reply:
Hmm I see yeah this `Clone` bound is annoying... You're right let's keep this type in there...

Reply:
Yeah :/

Reply:
Just make sure it gets visited in the folders etc